File: AdvancePython/Day-3 Advanced OOP concepts/Task-3-1.py
# Task 3:
# 1. Inheritance:
class Animal:
    def __init__(self, name: str) -> None:
        self.name: str = name

    def info(self) -> None:
        print(f"Animal name: {self.name}")


class Dog(Animal):
    def sound(self) -> None:
        print(f"{self.name} barks")


dog = Dog("Buddy")
dog.info()
dog.sound()

File: node.js/Day-3/Task-3-1.js
// Task 3:
// 1. Inheritance:

class Animal {
    constructor(name) {
        this.name = name;
    }

    info() {
        console.log(`Animal name: ${this.name}`);
    }
}

class Dog extends Animal {
    constructor(name) {
        super(name);    // call the parent class
    }

    sound() {
        console.log(`${this.name} Barks..`);
    }
}

const dog = new Dog("Tommy");
dog.info();
dog.sound();